include/highfive/H5File.hpp

```diff
@@ -111,6 +111,9 @@ class File: public Object, public NodeTraits<File>, public AnnotateTraits<File>
         return details::get_plist<FileAccessProps>(*this, H5Fget_access_plist);
     }
 
+  protected:
+    File() = default;
+
```

Comment:
This protected default constructor is needed for MSVC which otherwise complains that there is no suitable default constructor.

Reply:
Can that be a comment in the source?

Reply:
I'm not too convinced. My impression is that we don't comment on this type of thing:
* https://github.com/BlueBrain/HighFive/blob/master/include/highfive/H5Object.hpp#L71
* https://github.com/BlueBrain/HighFive/blob/master/include/highfive/H5Group.hpp#L31
* https://github.com/BlueBrain/HighFive/blob/master/include/highfive/H5DataSpace.hpp#L108

Instead we have CI that checks if it builds on various platforms. The comment is difficult to write in a precise way, because I don't have easy access to MS compilers other than through CI. Additionally, in a year it might work with MSVC, just not with the version we use in CI.